datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/domain/CreateDomainResolver.java

```diff
@@ -0,0 +1,91 @@
+package com.linkedin.datahub.graphql.resolvers.domain;
+
+import com.google.common.collect.ImmutableList;
+import com.linkedin.data.template.SetMode;
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.authorization.AuthorizationUtils;
+import com.linkedin.datahub.graphql.authorization.ConjunctivePrivilegeGroup;
+import com.linkedin.datahub.graphql.authorization.DisjunctivePrivilegeGroup;
+import com.linkedin.datahub.graphql.exception.AuthorizationException;
+import com.linkedin.datahub.graphql.generated.CreateDomainInput;
+import com.linkedin.domain.DomainProperties;
+import com.linkedin.entity.client.EntityClient;
+import com.linkedin.events.metadata.ChangeType;
+import com.linkedin.metadata.Constants;
+import com.linkedin.metadata.authorization.PoliciesConfig;
+import com.linkedin.metadata.key.DomainKey;
+import com.linkedin.metadata.utils.GenericAspectUtils;
+import com.linkedin.mxe.MetadataChangeProposal;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import static com.linkedin.datahub.graphql.resolvers.ResolverUtils.*;
+
+/**
+ * Resolver used for creating a new Domain on DataHub. Requires the MANAGE_DOMAINS privilege.
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class CreateDomainResolver implements DataFetcher<CompletableFuture<String>> {
+
+  private final EntityClient _entityClient;
+
+  @Override
+  public CompletableFuture<String> get(DataFetchingEnvironment environment) throws Exception {
+
+    final QueryContext context = environment.getContext();
+    final CreateDomainInput input = bindArgument(environment.getArgument("input"), CreateDomainInput.class);
+
+    return CompletableFuture.supplyAsync(() -> {
+
+      if (!isAuthorizedToCreateDomain(context)) {
+        throw new AuthorizationException("Unauthorized to perform this action. Please contact your DataHub administrator.");
+      }
+
+      // TODO: Add exists check. Currently this can override previously created domains.
```

Comment:
you can take a look at how we do this in the add/create tag modal to prevent someone from creating an existing tag name

Reply:
I plan to do this (and delete domain) in a followup based on demand